board.js

```diff
@@ -67,85 +80,108 @@ export default class Board {
       // this.ctx.beginPath();
       this.ctx.moveTo(0.5 + i + this.padding, this.padding);
       this.ctx.lineTo(0.5 + i + this.padding, this.height + this.padding);
       this.ctx.strokeStyle = this.strokeStyle;
+      
+      /* Keep the first and last lines of the grid opaque */
+      if ( i > 0 && i < this.height )
+        this.ctx.globalAlpha = this.gridOpacity;
+
       this.ctx.stroke();
+      this.ctx.globalAlpha = 1.0;
 
       const num = i / this.gridsize;
       if (num < 1) continue;
+
       this.ctx.beginPath();
-      this.ctx.fillStyle = "slategray";
-      this.ctx.textAlign = "center";
-      this.ctx.textBaseline = "middle";
       const character = num > 26 ? String.fromCharCode(num + 70) : String.fromCharCode(num + 64)
+
       this.ctx.fillText(
         character,
         this.padding + i - this.gridsize / 2,
-        this.padding - 5
+        this.padding - 8
       );
     }
-
+    /* Drawing the numeral coordinate markers */
     for (let i = 0; i <= this.height; i += this.gridsize) {
       this.ctx.moveTo(this.padding, 0.5 + i + this.padding);
       this.ctx.lineTo(this.width + this.padding, 0.5 + i + this.padding);
       this.ctx.strokeStyle = this.strokeStyle;
+      
+      /* Keep the first and last lines of the grid opaque */
+      if ( i > 0 && i < this.height )
+        this.ctx.globalAlpha = this.gridOpacity;
+      
       this.ctx.stroke();
+      this.ctx.globalAlpha = 1.0;
 
       this.ctx.beginPath();
-      this.ctx.fillStyle = "slategray";
-      this.ctx.textAlign = "center";
-      this.ctx.textBaseline = "middle";
       const num = i / this.gridsize;
       if (num < 1) continue;
+
       this.ctx.fillText(
         String(num),
         this.padding - 7,
         this.padding + i - this.gridsize / 2
       );
     }
 
+    /* Drawing the scale marker */
     this.ctx.beginPath();
-    this.ctx.fillStyle = "slategray";
     this.ctx.fillText(
       "1 square = 5ft",
       this.width - this.padding - 10,
       this.height + this.padding + 7
     );
+  }
+
+  draw() {
+    this.ctx.beginPath();
+    this.ctx.fillStyle = this.darkMode ? fillDarkMode : fillLightMode;
+    this.ctx.fillRect(
+      0,
+      0,
+      this.width * this.gridsize,
+      this.height * this.gridsize
+    );
+
+    if (this.background) {
+      const img = new Image();
+
+      img.onload = () => {
+
+        /* We don't want to scale images because we're assuming that any 
+           default maps or user-provided maps meet the specifications we 
+           outlined in the README.
+           Instead of scaling, trim provided image to the map */
+        this.ctx.drawImage(img, 
+                          0, 0, this.width, this.height, /* Clip image */
+                          this.padding, this.padding, this.width * this.zoom, this.height * this.zoom); /* Draw Image */
+      };
+      img.onerror = (err) => {
+        throw err;
+      };
+      img.src = this.background;
+    }
+
+    this.drawText();
```

Comment:
Looking at it again, this function name probably needs to be updated. It draws the gridlines along with the text, so this name is lying a bit